### PR TITLE
Provide simple mechanism for building JMC under Docker

### PR DESCRIPTION
**Pull request by willthames**
Use docker-compose to create appropriate containers for
following the build instructions, and put the result in
the target directory.

Instructions are provided

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -276,6 +276,15 @@ The above will not run StacktraceModelTest, as that is also matched by "test.exc
 
 Note that if UI-tests are supposed to be part of the filtered run the "uitests" profile needs to be specified as well. Otherwise the UI won't start up and so the tests fail.
 
+
+## Building using docker and docker-compose
+
+```
+docker-compose run -f docker/docker-compose.yml
```
Comment (klaraward) [inline README.md]: I think this command is not complete? It's needs a service to run as well?

File: README.md [hunk 2]
```diff
@@ -276,6 +276,15 @@ The above will not run StacktraceModelTest, as that is also matched by "test.exc
...
+```
+
+Once build has finished the results will be in the `target` directory
```
Comment (klaraward) [inline README.md]: For me it ended up in target/target

**Comment by klaraward** [comment]
This is awesome :)

File: README.md (hunk 1 above)
Comment (willthames) [reply README.md]: Thanks, good spot

File: README.md (hunk 2 above)
Comment (willthames) [reply README.md]: Fixed now, thank you

File: README.md [hunk 3]
```diff
@@ -276,6 +276,15 @@ The above will not run StacktraceModelTest, as that is also matched by "test.exc
...
 Note that if UI-tests are supposed to be part of the filtered run the "uitests" profile needs to be specified as well. Otherwise the UI won't start up and so the tests fail.
 
+
+## Building using docker and docker-compose
+
+```
+docker-compose run -f docker/docker-compose.yml jmc
+```
+
+Once build has finished the results will be in the `target` directory
```
Comment (klaraward) [inline README.md]: For me, it's this command that works- `docker-compose -f docker/docker-compose.yml run jmc`

Comment (willthames) [reply README.md]: That's effectively identical isn't it - just changing the order between `run` and `-f docker/docker-compose.yml` ?

Comment (willthames) [reply README.md]: You were definitely correct that I'd missed `jmc` at the end previously

Comment (klaraward) [reply README.md]: I guess it should be the same but having run before the file does not work for me, maybe it's just the version of docker-compose I have (docker-compose version 1.23.2, build 1110ad01)

Comment (willthames) [reply README.md]: Fair enough, I have no objection to changing that!

Comment (willthames) [reply README.md]: I also might be wrong that it ever worked for me and just wrote what I thought I'd run.

It's fixed now though

**Comment by thegreystone** [comment]
@willthames - we can't accept contributions to the JMC project without a signed OCA, just like contributions to Eclipse needs an ECL etc. It's a very simple procedure. Let me know if you need any assistance.

**Comment by willthames** [comment]
@thegreystone sorry, I missed your earlier comment there. It looks fairly straightforward so I'll try and sort out the paperwork in the next few days

**Comment by thegreystone** [comment]
> @thegreystone sorry, I missed your earlier comment there. It looks fairly straightforward so I'll try and sort out the paperwork in the next few days

Excellent! Thanks a bunch! :)

**Comment by willthames** [comment]
@thegreystone my OCA came through the other day - I'm on the list now.

**Comment by thegreystone** [comment]
Excellent! Thanks Will!

**Comment by thegreystone** [comment]
Was a bit too quick - JMC should be built with JDK 8 for now, and I run into some trouble unless I switch to using JDK 8. @willthames - do you mind if I amend the final check-in into the mercurial repo by instead doing FROM openjdk:8-jdk-stretch?

**Comment by willthames** [comment]
@thegreystone whatever works for you but this works perfectly for me under JDK 11 - what is the issue that needs this to be built under JDK 8?

**Comment by thegreystone** [comment]
Hm. Weird, we never got the tooling to work properly under JDK 11, but perhaps the updates to the latest and greatest make it easier. For me it failed. Perhaps we can take a closer look later, but core is intended to be built with JDK 7 (still used by code at Oracle that _think_ still runs on JDK 7 for certain services), and the app is intended to be runnable in Eclipse running on a JDK 8.

Either way, thank you for the contribution!

**Pull request merged**